darwin/dataset/remote_dataset_v2.py

```diff
@@ -718,13 +732,18 @@ def register_single_slotted(
                 "storage_slug": object_store.name,
             }
             print(f"Registering {len(chunk)} items...")
-            response = self.client.api_v2.register_items(payload, team_slug=self.team)
-            for item in json.loads(response.text)["items"]:
-                item_info = f"Item {item['name']} registered with item ID {item['id']}"
-                results["registered"].append(item_info)
-            for item in json.loads(response.text)["blocked_items"]:
-                item_info = f"Item {item['name']} was blocked for the reason: {item['slots'][0]['reason']}"
-                results["blocked"].append(item_info)
+            try:
+                response = self.register_items_with_retry(payload, team_slug=self.team)
```

Comment:
It would be more conventional for this line should be left unchanged and instead have the retry logic implementation in the client code. Changes are more places would use that client (potentially clients use it directly in their scripts) and then they would benefit from the retry mechanis,

Reply:
Great point, have made this change